File: app/src/main/java/com/smedialink/abakarmagomedov/githubapplication/di/AppComponent.java
package com.smedialink.abakarmagomedov.githubapplication.di;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by abakarmagomedov on 15/05/17.
 */

@Singleton
@Component(modules = {AppModule.class, NetModule.class, UtilsModule.class})
public interface AppComponent {

    UsersComponent createUsersComponent();
    UserPostComponent createUserPostComponent();

}

File: app/src/main/java/com/smedialink/abakarmagomedov/githubapplication/di/AppComponent.kt
package com.smedialink.abakarmagomedov.githubapplication.di

import dagger.Component
import javax.inject.Singleton

/**
 * Created by abakarmagomedov on 15/05/17.
 */
@Singleton
@Component(modules = [AppModule::class, NetModule::class, UtilsModule::class])
interface AppComponent {
    fun createUsersComponent(): UsersComponent
    fun createUserPostComponent(): UserPostComponent
}
